### PR TITLE
:construction: 21 train logistic regression  data preparation for machine learning

### LINKED ISSUE #21: Train Logistic Regression : Data Preparation for Machine Learning
# Data Preparation

## Description

As a user, I want to perform several tasks on a CSV file in order to prepare the data for logistic regression modeling.

## Acceptance Criteria:
- [x]  Load the CSV file containing the data.
- [x] Remove columns that are not relevant for analysis.
- [x] Encode categorical variables to make them usable for modeling.
- [x] Check and handle missing values in the data to avoid biases.
- [x] Separate the features from the target variable.
- [x] Split the data into two distinct sets: a training set and a test set, to evaluate the model's performance.

### PR DESCRIPTION
**Pull request by farinaleo**
Add feature to correctly format the csv file with a config file.
- apply MI for missing value
- apply Z-score to normalise data
- remove useless columns

config file example:
```config
[COLUMNS] # If False drop column
Index = False
First Name = False
Last Name = False
Birthday = False
Best Hand = True
Care of Magical Creatures = False
Astronomy = False
Arithmancy = False
[REPLACE:Hogwarts House] # replace value from a specific column
Gryffindor = 0
Ravenclaw = 1
Slytherin = 2
Hufflepuff = 3
[REPLACE:Best Hand]
Left = 0
Right = 1
```

### CODE REVIEW

**Review by kazourak — APPROVED** [review]
nice
